modules/apps/commerce/commerce-order-content-web/src/main/java/com/liferay/commerce/order/content/web/internal/fragment/renderer/HeaderFragmentRenderer.java

```diff
@@ -0,0 +1,234 @@
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.security.permission.resource.PortletResourcePermission;
+import com.liferay.portal.kernel.theme.PortletDisplay;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.Constants;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.kernel.util.WebKeys;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+
+import java.util.Locale;
+
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletContext;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Brian I. Kim
+ */
+@Component(service = FragmentRenderer.class)
+public class HeaderFragmentRenderer implements FragmentRenderer {
+
+	@Override
+	public String getCollectionKey() {
+		return "commerce-product";
+	}
+
+	@Override
+	public String getLabel(Locale locale) {
+		return _language.get(locale, "header");
+	}
+
+	@Override
+	public void render(
+			FragmentRendererContext fragmentRendererContext,
+			HttpServletRequest httpServletRequest,
+			HttpServletResponse httpServletResponse)
+		throws IOException {
+
+		/*				Object infoItem = httpServletRequest.getAttribute(
+							InfoDisplayWebKeys.INFO_ITEM);
+
+						if ((infoItem == null) || !(infoItem instanceof CommerceOrder)) {
+							if (_isEditMode(httpServletRequest)) {
+								_printPortletMessageInfo(
+									httpServletRequest, httpServletResponse);
+							}
+
+							return;
+						}*/
+
+		long commerceOrderId = ParamUtil.getLongValues(
+			httpServletRequest,
+			"_com_liferay_commerce_order_content_web_internal_portlet_" +
+				"CommerceOpenOrderContentPortlet_commerceOrderId")[0];
+
+		try {
+			if (_commerceOrderService.fetchCommerceOrder(commerceOrderId) ==
+					null) {
+
+				if (_isEditMode(httpServletRequest)) {
+					_printPortletMessageInfo(
+						httpServletRequest, httpServletResponse);
+				}
+
+				return;
+			}
+		}
+		catch (PortalException portalException) {
+			throw new RuntimeException(portalException);
+		}
+
+		try {
+			RequestDispatcher requestDispatcher =
+				_servletContext.getRequestDispatcher(
+					"/fragment/renderer/page.jsp");
+
+			HeaderFragmentDisplayContext headerFragmentDisplayContext =
+				new HeaderFragmentDisplayContext(
+					_commerceChannelLocalService, _commerceOrderEngine,
+					_commerceOrderHttpHelper, _commerceOrderService,
+					_commercePaymentIntegrationRegistry,
+					_commercePaymentMethodGroupRelLocalService,
+					_commercePaymentMethodRegistry, _configurationProvider,
+					httpServletRequest, _portal, _portletResourcePermission);
+
```

Comment:
Created a `HeaderFragmentDisplayContext` to handle passing in`HeaderActionModals` to `page.jsp`